shared/libraries/opcuatms/opcuatms_client/src/objects/tms_client_object_impl.cpp

```diff
@@ -77,32 +77,17 @@ void TmsClientObjectImpl::subscriptionStatusChangeCallback(UA_StatusChangeNotifi
     //TODO report on disconnect
 }
 
-uint32_t TmsClientObjectImpl::tryReadChildNumberInList(const std::string& nodeName)
+uint32_t TmsClientObjectImpl::readChildNumberInList(const std::string& nodeName)
 {
-    try
-    {
-        const auto childId = this->getNodeId(nodeName);
-        return tryReadChildNumberInList(childId);
-    }
-    catch(...)
-    {
-    }
-
-    return std::numeric_limits<uint32_t>::max();
+    const auto childId = this->getNodeId(nodeName);
+    return readChildNumberInList(childId);
 }
 
-uint32_t TmsClientObjectImpl::tryReadChildNumberInList(const opcua::OpcUaNodeId& nodeId)
+uint32_t TmsClientObjectImpl::readChildNumberInList(const opcua::OpcUaNodeId& nodeId)
 {
-    try
-    {
-        const auto numberInListId = clientContext->getReferenceBrowser()->getChildNodeId(nodeId, "NumberInList");
-        return VariantConverter<IInteger>::ToDaqObject(client->readValue(numberInListId));
-    }
-    catch(...)
-    {
-    }
-
-    return std::numeric_limits<uint32_t>::max();
+    const auto numberInListId = clientContext->getReferenceBrowser()->getChildNodeId(nodeId, "NumberInList");
+    const auto variant = clientContext->getAttributeReader()->getValue(numberInListId, UA_ATTRIBUTEID_VALUE);
+    return VariantConverter<IInteger>::ToDaqObject(variant);
```

Comment:
Will this now crash if the NumberInList node is either missing or null?

Reply:
The main reason the "try" was there is for cases where someone does not implement NumberInList. I have a feeling this is not done in HBK currently. As such this was left there as a failsafe.

If we assume that the NumberInList is always configured, we also don't need to maintain the ordered/unordered lists.